From 5232c44ede66a11eacc1c3538c2383fed86c8b26 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Wed, 9 Sep 2015 19:34:21 -0700
Subject: [PATCH 078/296] StrictMode: Disable all strict mode functions when
 disable prop is set

Strict mode can still be used to produce red border flashes even when the
disable property is set, among other things.

Truly disable all strict-mode checks outright when the disable property
is set.
---
 core/java/android/os/StrictMode.java | 22 +++++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/core/java/android/os/StrictMode.java b/core/java/android/os/StrictMode.java
index 3b6df5df13a..8203373d989 100644
--- a/core/java/android/os/StrictMode.java
+++ b/core/java/android/os/StrictMode.java
@@ -1188,7 +1188,7 @@ public final class StrictMode {
 
         // For debug builds, log event loop stalls to dropbox for analysis.
         // Similar logic also appears in ActivityThread.java for system apps.
-        if (!doFlashes && (Build.IS_USER || suppress)) {
+        if ((!doFlashes && Build.IS_USER) || suppress) {
             setCloseGuardEnabled(false);
             return false;
         }
@@ -1478,6 +1478,11 @@ public final class StrictMode {
         // until the time the looper is idle again (right before
         // the next epoll_wait)
         void handleViolationWithTimingAttempt(final ViolationInfo info) {
+            final boolean suppress = SystemProperties.getBoolean(DISABLE_PROPERTY, false);
+            if (suppress) {
+                return;
+            }
+
             Looper looper = Looper.myLooper();
 
             // Without a Looper, we're unable to time how long the
@@ -1805,6 +1810,11 @@ public final class StrictMode {
      * @param policy the policy to put into place
      */
     public static void setVmPolicy(final VmPolicy policy) {
+        final boolean suppress = SystemProperties.getBoolean(DISABLE_PROPERTY, false);
+        if (suppress) {
+            return;
+        }
+
         synchronized (StrictMode.class) {
             sVmPolicy = policy;
             sVmPolicyMask = policy.mask;
@@ -2028,6 +2038,11 @@ public final class StrictMode {
      */
     public static void onVmPolicyViolation(String message, Throwable originStack,
             boolean forceDeath) {
+        final boolean suppress = SystemProperties.getBoolean(DISABLE_PROPERTY, false);
+        if (suppress) {
+            return;
+        }
+
         final boolean penaltyDropbox = (sVmPolicyMask & PENALTY_DROPBOX) != 0;
         final boolean penaltyDeath = ((sVmPolicyMask & PENALTY_DEATH) != 0) || forceDeath;
         final boolean penaltyLog = (sVmPolicyMask & PENALTY_LOG) != 0;
@@ -2132,6 +2147,11 @@ public final class StrictMode {
      * we here read back all the encoded violations.
      */
     /* package */ static void readAndHandleBinderCallViolations(Parcel p) {
+        final boolean suppress = SystemProperties.getBoolean(DISABLE_PROPERTY, false);
+        if (suppress) {
+            return;
+        }
+
         // Our own stack trace to append
         StringWriter sw = new StringWriter();
         sw.append("# via Binder call with stack:\n");
-- 
2.11.0

